diracx-routers/src/diracx/routers/jobs/submission.py

```diff
@@ -0,0 +1,204 @@
+import logging
+from datetime import datetime, timezone
+from http import HTTPStatus
+from typing import Annotated
+
+from fastapi import Body, Depends, HTTPException
+from pydantic import BaseModel
+from typing_extensions import TypedDict
+
+from diracx.core.models import (
+    JobStatus,
+)
+from diracx.core.properties import JOB_ADMINISTRATOR, NORMAL_USER
+
+from ..auth import has_properties
+from ..dependencies import (
+    JobDB,
+    JobLoggingDB,
+)
+from ..fastapi_classes import DiracxRouter
+from ..utils.users import AuthorizedUserInfo, verify_dirac_access_token
+from .access_policies import ActionType, CheckWMSPolicyCallable
+
+logger = logging.getLogger(__name__)
+
+router = DiracxRouter(dependencies=[has_properties(NORMAL_USER | JOB_ADMINISTRATOR)])
+
+
+class InsertedJob(TypedDict):
+    JobID: int
+    Status: str
+    MinorStatus: str
+    TimeStamp: datetime
+
+
+class JobID(BaseModel):
+    job_id: int
+
+
+MAX_PARAMETRIC_JOBS = 20
+
+EXAMPLE_JDLS = {
+    "Simple JDL": {
+        "value": [
+            """Arguments = "jobDescription.xml -o LogLevel=INFO";
+Executable = "dirac-jobexec";
+JobGroup = jobGroup;
+JobName = jobName;
+JobType = User;
+LogLevel = INFO;
+OutputSandbox =
+    {
+        Script1_CodeOutput.log,
+        std.err,
+        std.out
+    };
+Priority = 1;
+Site = ANY;
+StdError = std.err;
+StdOutput = std.out;"""
+        ]
+    },
+    "Parametric JDL": {
+        "value": ["""Arguments = "jobDescription.xml -o LogLevel=INFO"""]
+    },
+}
+
+
+@router.post("/")
+async def submit_bulk_jobs(
+    job_definitions: Annotated[list[str], Body(openapi_examples=EXAMPLE_JDLS)],
+    job_db: JobDB,
+    job_logging_db: JobLoggingDB,
+    user_info: Annotated[AuthorizedUserInfo, Depends(verify_dirac_access_token)],
+    check_permissions: CheckWMSPolicyCallable,
+) -> list[InsertedJob]:
+    await check_permissions(action=ActionType.CREATE, job_db=job_db)
+
+    from DIRAC.Core.Utilities.ClassAd.ClassAdLight import ClassAd
+    from DIRAC.Core.Utilities.ReturnValues import returnValueOrRaise
+    from DIRAC.WorkloadManagementSystem.Service.JobPolicy import RIGHT_SUBMIT, JobPolicy
+    from DIRAC.WorkloadManagementSystem.Utilities.ParametricJob import (
+        generateParametricJobs,
+        getParameterVectorLength,
+    )
+
+    class DiracxJobPolicy(JobPolicy):
+        def __init__(self, user_info: AuthorizedUserInfo, allInfo: bool = True):
+            self.userName = user_info.preferred_username
+            self.userGroup = user_info.dirac_group
+            self.userProperties = user_info.properties
+            self.jobDB = None
+            self.allInfo = allInfo
+            self._permissions: dict[str, bool] = {}
+            self._getUserJobPolicy()
+
+    # Check job submission permission
+    policyDict = returnValueOrRaise(DiracxJobPolicy(user_info).getJobPolicy())
+    if not policyDict[RIGHT_SUBMIT]:
+        raise HTTPException(HTTPStatus.FORBIDDEN, "You are not allowed to submit jobs")
```

Comment:
Is it still needed now that we have the `check_permissions` function at line 80?

Reply:
Assuming `check_permissions` should do the right thing I removed this

Reply:
hmm, but now we get a failing test:

```
FAILED diracx-routers/tests/test_job_manager.py::test_user_without_the_normal_user_property_cannot_submit_job - AssertionError: [{'JobID': 1, 'MinorStatus': 'Job accepted', 'Status': 'Received', 'TimeStamp': '2024-12-14T19:45:24.667029Z'}]
assert 200 == <HTTPStatus.FORBIDDEN: 403>
 +  where 200 = <Response [200 OK]>.status_code
 +  and   <HTTPStatus.FORBIDDEN: 403> = HTTPStatus.FORBIDDEN
```